### PR TITLE
[UI/UX:InstructorUI] Clarify poll edit confirmation

### LINKED ISSUE #9090: Edit poll button description
**What problem are you trying to solve with Submitty**
Currently, the save change button in Instructor's edit poll page is confusing(red circle in the screenshot)
screenshot

**Describe the way you'd like to solve this problem**
change "Edit Poll" to "Save Change"

### PR DESCRIPTION
**Pull request by AdamTuhacek**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
In the edit poll screen, edit is listed as the title as well as the button to save the changes, which can be misleading.

### What is the new behavior?
Change the confirmation button to save changes.

Fixes #9090

### CODE REVIEW

**Review by shailpatels — APPROVED** [review]
Looks good, here is a screenshot of the UI changes

image

**Comment by hansongu123** [comment]
Personally, I like just "Save", or at least capitalize "changes" to fit with the style of other buttons.

**Comment by AdamTuhacek** [comment]
> Personally, I like just "Save", or at least capitalize "changes" to fit with the style of other buttons.

Yeah, I corrected it to have the C capitalized. I'm not sure whether save/save changes would make much of a difference from the user perspective, I think either makes more sense than before.

**Comment by hansongu123** [comment]
"Save Poll", since it is "Edit Poll" before?

**Comment by hansongu123** [comment]
In the `polls.spec.js` file, also change the part where the automation test is invoking the button by the text on it, for example: 
image
to make sure it passes the test.

**Comment by AdamTuhacek** [comment]
> In the `polls.spec.js` file, also change the part where the automation test is invoking the button by the text on it, for example: image to make sure it passes the test.

Ok, thanks! I updated the name to save poll and changed the cypress test case.

**Comment by hansongu123** [comment]
Also on `line 289`, do the same change to pass the e2e test. Make sure you're changing it for every occurrence of the button text being checked.

File: site/app/templates/polls/PollForm.twig [hunk 1]
```diff
@@ -163,7 +163,7 @@
         <span id="empty-response-error" class="red-message polls-submit-error" hidden>Please make sure all responses are filled in.<br/></span>
         <span id="file-size-error" class="red-message polls-submit-error" hidden>Please make sure to upload a file that does not exceed the maximum size.<br/></span>
         <a href="{{base_url}}" class="btn btn-danger">Cancel</a>
-        <button type="submit" id="poll-form-submit" class="btn btn-primary" {{ poll is null ? 'disabled' }}>{{ poll is null ? 'Add' : 'Edit' }} Poll</button>
+        <button type="submit" id="poll-form-submit" class="btn btn-primary" {{ poll is null ? 'disabled' }}>{{ poll is null ? 'Add' : 'Save Poll' }} </button>
```
Comment (hansongu123) [inline site/app/templates/polls/PollForm.twig:166]: Also revert the irrelevant structural change on how this button's text is formulated. Only change the part where it says 'Edit'.

Comment (AdamTuhacek) [reply site/app/templates/polls/PollForm.twig]: Ok, thank you very much for the detailed feedback! Changes are made

File: site/app/templates/polls/PollForm.twig [hunk 2]
```diff
@@ -163,7 +163,7 @@
         <span id="empty-response-error" class="red-message polls-submit-error" hidden>Please make sure all responses are filled in.<br/></span>
         <span id="file-size-error" class="red-message polls-submit-error" hidden>Please make sure to upload a file that does not exceed the maximum size.<br/></span>
         <a href="{{base_url}}" class="btn btn-danger">Cancel</a>
-        <button type="submit" id="poll-form-submit" class="btn btn-primary" {{ poll is null ? 'disabled' }}>{{ poll is null ? 'Add' : 'Edit' }} Poll</button>
+        <button type="submit" id="poll-form-submit" class="btn btn-primary" {{ poll is null ? 'disabled' }}>{{ poll is null ? 'Add' : : 'Save' }} Poll </button>
```
Comment (williamjallen) [inline site/app/templates/polls/PollForm.twig:166]: Is the extra colon a typo here?

```suggestion
        <button type="submit" id="poll-form-submit" class="btn btn-primary" {{ poll is null ? 'disabled' }}>{{ poll is null ? 'Add' : 'Save' }} Poll </button>
```

Comment (hansongu123) [reply site/app/templates/polls/PollForm.twig:166]: Looks like it

Comment (AdamTuhacek) [reply site/app/templates/polls/PollForm.twig]: Oops, yes, thanks for catching that.

**Comment by hansongu123** [comment]
Sorry, I got confused when you mentioned about Polls' Cypress tests, and I just took a closer look. You shouldn't be modifying the `polls.spec.js` file because there isn't an element invoking that involves getting the content string of the changed text, instead when adding or saving changes to a poll, js would just invoke the button by looking up its type (for example in line 293). You should be good if you revert all the changes you made in the Cypress file (passes locally when I did it). Give it a local test, and don't forget to do a `submitty_install_site` to update the change. Also keep an eye on the duplicated polls named "Poll Cypress Test" if you failed to make it to the end locally and it doesn't get properly deleted.

File: site/cypress/e2e/polls.spec.js [hunk 3]
```diff
@@ -241,7 +241,7 @@ describe('Test cases revolving around polls functionality', () => {
         cy.visit(['sample', 'polls']);
         cy.contains('Poll Cypress Test').siblings(':nth-child(1)').children().click();
         cy.url().should('include', 'sample/polls/editPoll');
-        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Edit Poll');
+        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Save Poll');
```
Comment (hansongu123) [inline site/cypress/e2e/polls.spec.js:244]:
```suggestion
        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Edit Poll');
```

File: site/cypress/e2e/polls.spec.js [hunk 4]
```diff
@@ -286,7 +286,7 @@ describe('Test cases revolving around polls functionality', () => {
         cy.contains('Poll Cypress Test').siblings(':nth-child(6)').children().click();
         cy.contains('Poll Cypress Test').siblings(':nth-child(1)').children().click();
         cy.url().should('include', 'sample/polls/editPoll');
-        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Edit Poll');
+        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Save Poll');
```
Comment (hansongu123) [inline site/cypress/e2e/polls.spec.js:289]:
```suggestion
        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Edit Poll');
```

**Review by hansongu123 — COMMENTED** [review]
These changes only change the check of the header in the poll editing page. It is really confusing indeed since the text is the same as in that button in `PollForm.twig`

**Comment by AdamTuhacek** [comment]
> These changes only change the check of the header in the poll editing page. It is really confusing indeed since the text is the same as in that button in `PollForm.twig`

Awesome, thanks for all the help Hanson!

**Review by williamjallen — APPROVED** [review]

**Pull request merged**